src/go/supervisor/pkg/svc/supervisor.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright © 2021 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package svc
+
+import (
+	"fmt"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/supervisor/pkg/port"
+)
+
+type supervisor struct {
+	eventLogger debug.Log
+	portServer  *port.Server
+}
+
+func NewSupervisor(portServer *port.Server, logger debug.Log) *supervisor {
+	return &supervisor{
+		eventLogger: logger,
+		portServer:  portServer,
+	}
+}
+
+// Execute is the core of the supervisor service to handle all
+// the service related event requests. Any outside function
+// calls MUST be called in a goroutine.
+// The signature must NOT change since it is part of the standard
+// service handler interface
+// https://github.com/golang/sys/blob/1c4a2a72c664c42691e45fe3a31510ec5ba13cfb/windows/svc/service.go#L115
+func (s *supervisor) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
```

Comment:
What is `ssec`? That name is unclear.
(Also, consider dropping the use of named return values; that is typically confusing.)